### PR TITLE
Step 2C includes controls with value that are directly referenced by labelledby or describedby

### LINKED ISSUE #179: Step 2E isn't used for form field + name-from-label when it is the current node in an aria-labelledby traversal
Using the accName for the button in the following markup as a reduced example:

```html
<label for="inputId">Something</label>
<input id="inputId">
<button aria-labelledby="inputId">***</button>
```

My reading of the accname spec is that the following should happen for the button:
1. It should hit 2.B.ii and follow the idRef `inputId`
2. Start from step 2 with the input as the current node
3. The input should use step 2E to calculate its text alternative, and return the text of the `<label>`
4. The button's accName should be "Something"

This is not the case for Firefox, Chrome, Edge, or Safari in my testing. All of them return the value of the input as its text alternative when within an `aria-labelledby` traversal instead of the name derived from the `<label>`. My guess is that browsers are following step 2C here, even though the input is directly referenced, and not within a parent label.

I'd like to confirm whether the correct behavior is for the button to have the accName "Something" here, or whether the current browser behavior is correct.

### PR DESCRIPTION
**Pull request by smhigley**
Closes #179, based on discussed resolution

Updates the wording of 2C to specify returning the value text of all controls that allow users to control their value, whether or not they are embedded within a referenced label, or are the referenced label. This is the current real-world behavior of all tested browsers.

I also included controls that are part of an `aria-describedby` calculation, since in testing FF, Chrome, Edge, and Safari, all browsers implement this in practice as well (ref: https://jsfiddle.net/tfogpnvj/ for a reduced example).

# PR tracking
Check these once you have confirmed the related change is not necessary (also add N/A) OR
when the relevant issue or PR has been made. Leave uncheck if you are unsure. Read the
[Process Document](https://github.com/w3c/aria/wiki/ARIA-WG-Process-Document/_edit) or
[Test Overview](https://github.com/w3c/aria/wiki/ARIA-Test-Overview) for more information.

* [x] Related Core AAM Issue/PR: n/a
* [x] Related AccName Issue/PR: n/a
* [x] Related APG Issue/PR: n/a
* [x] Any other dependent changes? n/a

# Implementation tracking

* [x] validator tests n/a
* [x] WPT tests: n/a (not requiring accname test, yet!)
* [x] Browser implementations (link to issue or when done, link to commit):
   * WebKit: n/a, matches current implementation
   * Gecko: n/a, matches current implementation
   * Blink: n/a, matches current implementation
* [x] Does this need AT implementations?

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/accname/pull/183.html" title="Last updated on Nov 3, 2023, 11:03 PM UTC (85159e4)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/accname/183/681094c...85159e4.html" title="Last updated on Nov 3, 2023, 11:03 PM UTC (85159e4)">Diff</a>

### CODE REVIEW

File: index.html
Comment (cookiecrook) [reply index.html:392]: Accurate, but possibly less clear to the reader? Note: I don't have an opinion for or against this suggestion.

**Review by cookiecrook — APPROVED** [review]
Looks good, but please don't merge this one until after #186 is merged.

File: index.html
Comment (MelSumner) [reply index.html:392]: What do you think about having both ? `embedded control (current node)` has a role,...`

I've gotten used to "embedded control" and what it means within the context of this specification, so I would hesitate to be enthusiastic about removing it.

**Review by accdc — APPROVED** [review]

**Comment by spectranaut** [comment]
@MelSumner I think this is ready for merge! Not waiting for any dependent changes.

**Review by adampage — APPROVED** [review]

**Comment by MelSumner** [comment]
@smhigley could you rebase? Once it's green we're ready to merge.

**Comment by MelSumner** [comment]
👋  @smhigley We'd love to get this rebased for a merge, would you be able to do that?

**Comment by MelSumner** [comment]
This PR has been moved to the mono repo: https://github.com/w3c/aria/pull/2210

**Pull request closed without merging**
